sklearn/datasets/tests/test_samples_generator.py

```diff
@@ -262,13 +262,23 @@ def test_make_blobs_n_samples_list_with_centers():
     X, y = make_blobs(n_samples=n_samples, centers=centers,
                       cluster_std=cluster_stds, random_state=0)
 
-    assert_equal(X.shape, (sum(n_samples), 2), "X shape mismatch")
-    assert_equal(y.shape, (sum(n_samples),), "y shape mismatch")
-    assert_equal(np.unique(y).shape, (3,), "Unexpected number of blobs")
+    assert X.shape == (sum(n_samples), 2), "X shape mismatch"
+    assert all(np.bincount(y, minlength=len(n_samples)) == n_samples), \
+        "Incorrect number of samples per blob"
     for i, (ctr, std) in enumerate(zip(centers, cluster_stds)):
         assert_almost_equal((X[y == i] - ctr).std(), std, 1, "Unexpected std")
 
 
+def test_make_blobs_n_samples_list_centers_none():
```

Comment:
```
@pytest.mark.parametrize(
    "n_samples",
    [[5, 3, 0],
     np.array([5, 3, 0]),
     tuple([5, 3, 0])]
)
def test_make_blobs_n_samples_centers_none(n_samples):
...
```
@glemaitre When using parametrize here, the test fails and I don't get the reason.
Would you know why? I had a look at how it is used in other test modules, but I can't find any explanation.

`test_make_blobs_n_samples_centers_none() missing 1 required positional argument: 'n_samples'`

Reply:
are you running the tests with pytest?

Reply:
@jnothman I was using nosetest my bad. Works fine w/ pytest.